ZFIN 7771: NTR Edit (PROTOTYPE)

rtaylorzfin opened the pull request:
Christian, I fully expect to need to change the persistence strategy on this. For now, I'm using the GenomeLocation hibernate entity just to get a prototype working, but I am planning on changing it after getting your feedback. I'd be interested in hearing other advice you have for how to improve the structure of the code here.  Also, I'm not sure about the way I condition the display of "Chromosomal Location" on the type of the marker in the gene-edit.jsp.

Thanks!
Ryan

cmpich commented on source/org/zfin/marker/presentation/MarkerChromosomalLocationController.java:
```diff
@@ -0,0 +1,94 @@
+
+@Controller
+@RequestMapping("/marker")
+public class MarkerChromosomalLocationController {
```
I would use 
@RestController
for this class and then use JsonView to configure the response JSON.

cmpich commented on source/org/zfin/marker/presentation/MarkerChromosomalLocationController.java:
```diff
@@ -0,0 +1,94 @@
+                                            @PathVariable Long ID,
+                                            @Valid @RequestBody ChromosomalLocationBean chromosomalLocationBean) {
+
+        GenomeLocation genomeLocation = markerRepository.getGenomeLocationByID(ID);
```
Do we need to add an evidence code to each GenomeLocation? Need to ask Leyla...

cmpich commented on source/org/zfin/marker/repository/HibernateMarkerRepository.java:
```diff
@@ -814,6 +815,50 @@ public void addMarkerRelationshipAttribution(MarkerRelationship mrel, Publicatio
     }
 
+
+    public GenomeLocation addGenomeLocation(GenomeLocation genomeLocation) {
```
I am not sure if you can persist new / updated GenomeLocation objects. That class maps to 
sequence_feature_chromosome_location_generated
and records in there may get overwritten. We need to check the DB functions that are responsible for getting data from sequence_feature_chromosome_location to sequence_feature_chromosome_location_generated

rtaylorzfin replied:
I ran the jenkins job (regenerate-chromosome-mart) on the cell vm and found that it did not overwrite the data I am storing in that warehouse table. This makes sense when digging into that job because the delete statements all seem to be conditional on the source column, like shown below.  However, I'm not sure if that job is even run anymore?

```
delete from sequence_feature_chromosome_location_generated
 where sfclg_location_source = 'ZfinGbrowseStartEndLoader';
```

cmpich commented on source/org/zfin/profile/presentation/ChromosomalLocationBean.java:
```diff
@@ -0,0 +1,92 @@
+        return clBean;
+    }
+
+    public Long getID() {
```
use
@Setter
@Getter
instead of writing them by hand. Keeps the class smaller.

rtaylorzfin replied:
Done.

cmpich reviewed (commented):
See my comments.

rtaylorzfin commented:
Thank you! These comments are great. I'll clean it up according to your feedback. The persistence issue was one I knew would need to be changed. If you have some thoughts on the best way to store it, that would be great. It feels like there should already exist something, but if not, I can create a new table and mapping.

rtaylorzfin commented on home/javascript/react/containers/MarkerEditChromosomalLocation.js:
```diff
@@ -0,0 +1,173 @@
+
+const MarkerEditChromosomalLocation = ({
+    markerId,
+    type,
```
It's used for the heading, though maybe we could just use a hard-coded string.

rtaylorzfin commented on source/org/zfin/marker/repository/HibernateMarkerRepository.java:
```diff
@@ -814,6 +816,48 @@ public void addMarkerRelationshipAttribution(MarkerRelationship mrel, Publicatio
+    public MarkerLocation addMarkerLocation(MarkerLocation markerLocation) {
+        Session session = HibernateUtil.currentSession();
+        String savedValue = (String) session.save(markerLocation);
+        MarkerLocation savedEntity = (MarkerLocation) session.get(MarkerLocation.class, savedValue);
```
Okay, I'll check that the saved markerLocation gets an ID added by the session.save

rtaylorzfin commented on source/org/zfin/mapping.hbm.xml:
```diff
@@ -175,7 +175,7 @@
     </class>
 
-    <class name="org.zfin.mapping.FeatureLocation" table="sequence_feature_chromosome_location">
+    <class name="org.zfin.mapping.GenericChromosomalLocation" table="sequence_feature_chromosome_location">
```
Okay, will do.

rtaylorzfin commented:
Refactor "GenericChromosomalLocation" to "Location"

rtaylorzfin commented:
Christian, I addressed the issues raised here except for the one about moving from xml mapping to annotation mapping. I made an attempt to do that, but couldn't get it to work. Perhaps you could help me figure out the issues with that. I have a commit with my work-in-progress here:

https://github.com/rtaylorzfin/zfin/commit/d5bf619a644aff1cc4e218c8a37350ec39eaaa07

cmpich reviewed (approved):
Looks good. The migration of the Hibernate mapping migration we an do in a subsequent ticket. Please create such a ticket.

rtaylorzfin commented:
Created ticket https://zfin.atlassian.net/browse/ZFIN-7890
